Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/storage/repo/DefaultContactDiaryRepository.kt

```diff
@@ -0,0 +1,205 @@
+    }
+
+    override suspend fun updatePerson(contactDiaryPerson: ContactDiaryPerson) {
+        Timber.d("Updating person $contactDiaryPerson")
+        val contactDiaryPersonEntity = contactDiaryPerson.toContactDiaryPersonEntity()
+        executeWhenIdNotDefault(contactDiaryPersonEntity.personId) {
+            contactDiaryPersonDao.update(contactDiaryPersonEntity)
+        }
+    }
+
+    override suspend fun deletePerson(contactDiaryPerson: ContactDiaryPerson) {
+        Timber.d("Deleting person $contactDiaryPerson")
+        val contactDiaryPersonEntity = contactDiaryPerson.toContactDiaryPersonEntity()
+        executeWhenIdNotDefault(contactDiaryPersonEntity.personId) {
+            contactDiaryPersonDao.delete(contactDiaryPersonEntity)
+        }
+    }
+
+    override suspend fun deletePeople(contactDiaryPeople: List<ContactDiaryPerson>) {
+        Timber.d("Deleting people $contactDiaryPeople")
+        val contactDiaryPersonEntities = contactDiaryPeople
+            .map {
+                val contactDiaryPersonEntity = it.toContactDiaryPersonEntity()
+                executeWhenIdNotDefault(contactDiaryPersonEntity.personId)
+                return@map contactDiaryPersonEntity
+            }
+        contactDiaryPersonDao.delete(contactDiaryPersonEntities)
+    }
+
+    override suspend fun deleteAllPeople() {
+        Timber.d("Clearing contact diary person table")
+        contactDiaryPersonDao.deleteAll()
+    }
+
+    // Person encounter
+    private suspend fun List<ContactDiaryPersonEncounterEntity>.toContactDiaryPersonEncounterSortedList():
+        List<ContactDiaryPersonEncounter> =
+        this.map {
+            val contactPerson = contactDiaryPersonDao.entityForId(it.id)
+            DefaultContactDiaryPersonEncounter(id = it.id, date = it.date, contactDiaryPerson = contactPerson)
+        }
+            .sortByNameAndIdASC()
+
+    override val personEncounters: Flow<List<ContactDiaryPersonEncounter>> =
+        contactDiaryPersonEncounterDao
+            .allEntries()
+            .map { it.toContactDiaryPersonEncounterSortedList() }
+
+    override fun personEncountersForDate(date: LocalDate): Flow<List<ContactDiaryPersonEncounter>> =
+        contactDiaryPersonEncounterDao
+            .entitiesForDate(date)
+            .map { it.toContactDiaryPersonEncounterSortedList() }
+
+    override suspend fun addPersonEncounter(contactDiaryPersonEncounter: ContactDiaryPersonEncounter) {
+        Timber.d("Adding person encounter $contactDiaryPersonEncounter")
+        val contactDiaryPersonEncounterEntity = contactDiaryPersonEncounter.toContactDiaryPersonEncounterEntity()
+        contactDiaryPersonEncounterDao.insert(contactDiaryPersonEncounterEntity)
+    }
+
+    override suspend fun deletePersonEncounter(contactDiaryPersonEncounter: ContactDiaryPersonEncounter) {
+        Timber.d("Deleting person encounter $contactDiaryPersonEncounter")
+        executeWhenIdNotDefault(contactDiaryPersonEncounter.id) {
```

Comment:
Can we do without this check?
What happens if we delete an ID that does not exist, could we just log an error?

What paths do you see that could lead to an ID being 0 here?

Reply:
Its a secure mechanism, maybe never needed but you never know.
I'm setting usually to 0, to trigger Room for autogenerating a ID.
If you would get a default id (e.g. Update Delete), then you would know something bad happened.